### DISCUSSION TITLE
Extend support to more file types

### DISCUSSION THREAD

**Discussion by farfilli** [rmcanany/SolidEdgeHousekeeper#181] [Ideas]
This is an important change in the application and needs to be planned with several methods and functions to be updated.
The idea is to extend support for more file types, even if they are not directly related to Solid Edge.

I would implement PDFs first; let's start with basic tasks like printing and watermarking.
The next batch would be to implement Word and Excel files and start with tasks like save as and print.
Another batch would manage step files, parasolids, and so on, and implement the conversion to Solid Edge files.

@rmcanany what do you think? Is it something you would consider or strictly Solid Edge files?

_Note: to be able to manage PDFs as I plan there is the need for an external DLL, I would choose from an open-source one; there are many so it will need some testing to select it. Something like:_

https://github.com/empira/PDFsharp-1.5
https://github.com/QuestPDF/QuestPDF

**Comment by rmcanany** [comment]
I appreciate the suggestion, but decided against processing non-SE files for now.  

I started to add a batch importer but it felt like a kludge to me.  I made a separate utility instead.

Closing as out of scope.
